rpc/common/moc_common_common.proto

```diff
@@ -112,6 +112,21 @@ message Status {
 	ValidationStatus validationStatus = 6;
 }
 
+// in the current implementation, cloudagent objects are generally identified by their group name 
+// and name. These fields are used to identify if a POST request should be treated as a create or 
+// an update
+message CloudSubResource {
```

Comment:
Let's move it to `moc_cloudagent_availabilityset.proto` and call it `VirtualMachineConfig` or something similar. I don't think this makes sense to place in common.proto if no other resource is going to use it.

Reply:
The intent is for a generic message that any subresource can be described in. It is not currently widely in use but it could be to replace all of the ad-hoc config id messages like NetworkInterface and Disk in the VirtualMachine.proto

Just trying to reduce repeated config messages, if you prefer that then I can change it.

Reply:
Your point is valid. Again it's the same discussion we had yesterday. Keeping it consistent across all the resources vs bringing in a minor improvement. In this case, I'd say let's keep things consistent so that people are not confused.